core/src/test/java/ai/timefold/solver/core/impl/heuristic/selector/move/generic/chained/SubChainSwapMoveTest.java

```diff
@@ -80,21 +77,21 @@ void oldAndNewTrailing() {
         var inverseVariableSupply = SelectorTestUtils.mockSingletonInverseVariableSupply(
                 new TestdataChainedEntity[] { a1, a2, a3, a4, a5, b1, b2, b3 });
 
-        try (var ephemeralMoveDirector = new MoveDirector<>(innerScoreDirector).ephemeral()) {
-            var scoreDirector = Mockito.spy(ephemeralMoveDirector.getScoreDirector());
-            var move = new SubChainSwapMove<>(variableDescriptor, inverseVariableSupply,
-                    new SubChain(Arrays.asList(a2, a3, a4)),
-                    new SubChain(Arrays.asList(b1, b2)));
-            move.doMoveOnly(scoreDirector);
-
-            SelectorTestUtils.assertChain(a0, a1, b1, b2, a5);
-            SelectorTestUtils.assertChain(b0, a2, a3, a4, b3);
-
-            verify(scoreDirector).changeVariableFacade(variableDescriptor, a2, b0);
-            verify(scoreDirector).changeVariableFacade(variableDescriptor, b3, a4);
-            verify(scoreDirector).changeVariableFacade(variableDescriptor, b1, a1);
-            verify(scoreDirector).changeVariableFacade(variableDescriptor, a5, b2);
-        }
+        var moveDirector = new MoveDirector<>(innerScoreDirector);
```

Comment:
I observed in several places that we are not invoking `executeTemporaryMove` when replacing the ephemeral director. Can this be a problem?

Reply:
It depends on what the test wants to do. Sometimes the test wants to do an undo, sometimes it doesn't. Since I'm not changing the behavior and the assertions still pass, that means I'm not breaking anything.